### PR TITLE
Automated cherry pick of #112806: test: demote service ClientIP affinity timeout tests from

### PR DESCRIPTION
**Pull request by dcbw**
Cherry pick of #112806 on release-1.25.

#112806: test: demote service ClientIP affinity timeout tests from

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Comment by dcbw** [comment]
@aojea @johnbelamaric @thockin @danwinship 
So here's a 1.25 backport. But does https://github.com/kubernetes/kubernetes/pull/112990#issuecomment-1275308631 indicate that it's not possible (either technically or by policy) to change existing release conformance tests?

**Comment by spiffxp** [comment]
>it's not possible (either technically or by policy) to change existing release conformance tests

Correct. I followed some breadcrumbs, there's a long slack thread linked in https://github.com/kubernetes/test-infra/pull/21082

Basically changing the conformance test metadata (inclusion/exclusion, name) after the fact messes with the CNCF conformance certification process.

Now it's been a while since we did that, let's say maybe the technical reasons resolved... I would still be extremely conservative about changing conformance tests post-release anyway.  What's the reasoning for the necessity of the backport?

**Comment by dcbw** [comment]
@spiffxp could you elaborate on the tooling problems?

re "What's the reasoning for the necessity of the backport?" see https://github.com/kubernetes/kubernetes/pull/112806#issue-1392654033 . We've all agreed this test is bad and should be demoted, because a number of 3rd party network plugins have trouble implementing it exactly the same way that iptables/ipvs do (which don't implement it the same way either!).

I'm happy to dedicate some resources to working through the problem, I just need to know what to help with :)

**Comment by dcbw** [comment]
@spiffxp I think the interesting bit is https://kubernetes.slack.com/archives/C78F00H99/p1613618390083100?thread_ts=1613501041.053200&cid=C78F00H99

```
I think it’s good that we tie together api/openapi-spec/swagger.json and tests/conformance/testdata/conformance.yaml (list of [Conformance] tests) in the same tree.The resulting e2e.test binary has those tests compiled in, and is picked up by Sonobouy based on the version of the cluster, which is used by most of the community to create logs for submission to cncf/k8s-conformance. The CNCF (using some APISnoop+prow automation) detects the k8s version in the sonobouy logs in PR/submission to cncf/k8s-conformance. The detected version informs which k/k branch we pull conformance.yaml from. We then ensure all required tests for that release are show up as passed in the logs.
```

But I'm struggling to understand (code pointers appreciated) how demoting a test would fail the "ensure all required tests for that release show up as passed", since conformance.yaml would no longer require the test that got demoted. Any idea here? I'm happy to dig into the CNCF tooling...

**Comment by dcbw** [comment]
@hh eg, wouldn't `GetMissingJunitTestsFromPRSuite()` from https://github.com/cncf-infra/verify-conformance/blob/main/pkg/suite/suite.go ignore tests in submitted logs, that aren't in conformance.yaml (because they were demoted)?

**Comment by dcbw** [comment]
@hh @spiffxp another ping; still interested in doing this...

**Comment by smarterclayton** [comment]
Removing a test from conformance because the community and technical leads agree the conformance test was too specific to an implementation should be sufficient for me to justify a change (and i am supportive of adding that to our formal conformance policy).

The only hypothetical scenario I have come up with where this would be bad is that a particular vendor decides an implementation is too expensive and refuses to update, thus impacting end users who depend on that feature.  However, if the community contributors and integrators agrees the features are too coupled, that problem does not arise.

**Comment by smarterclayton** [comment]
I opened https://github.com/kubernetes/community/pull/7071 to cover the change, sig-arch will review and then assuming we merge that we'll do quick round of consensus on the decision.  Ultimately the definition of conformance is owned by the project, and since we have agreed to demote the tests for release N I believe it is unreasonable to hold a conformant distribution in N from being conformant in N-1 when the cause is due to an oversight.

**Comment by abhat** [comment]
@hh @spiffxp the [changes covering the demotion process](https://github.com/kubernetes/community/pull/7071) are now merged. Can you please have another look at this PR and provide any inputs around @dcbw's comments?

**Comment by aojea** [comment]
> @aojea wanted a ping on this one too...

/lgtm
/approve

**Comment by johnbelamaric** [comment]
@Riaankl @heyste Can you answer the question regarding the conformance automation? Will this cause problems? If so, what needs to be done to fix it?

**Comment by heyste** [comment]
The conformance bot processes the results provided by sonobuoy which uses this [list of e2e tests](https://github.com/vmware-tanzu/sonobuoy/tree/main/cmd/sonobuoy/app/e2e/testLists) for each patch release of Kubernetes.

AFAIK, after the PR has merged, the sonobuoy team need to update the e2e files for 1.25 and then publish a new version of sonobuoy. I've asked for some clarification about this in the [sonobuoy slack channel](https://kubernetes.slack.com/archives/C6L3G051C/p1675394753827339).

**Comment by Riaankl** [comment]
@dcbw would this also be done for 1.24?
There is still submissions for [1.24 conformance PRs](https://github.com/cncf/k8s-conformance/pull/2404) that is affected by it.

**Comment by abhat** [comment]
@Riaankl yes indeed. https://github.com/kubernetes/kubernetes/pull/115510 is for the demotion of the test in 1.24

Thanks for the help here.

**Comment by abhat** [comment]
@Riaankl @heyste any chances of getting the clarity on conformance automation soon? it will be great if we can get this and #115510 landed before the Feb 10 monthly release cut-off.

**Comment by Riaankl** [comment]
@dcbw I have added this for discussion to the [SIG Arch meeting](http://bit.ly/sig-architecture) agenda for tomorrow.

**Comment by Riaankl** [comment]
This was discussed in SIG Arch meeting. From a conformance point of this PR can be merge. The only issue the that the is no clear tigger system to enforce the updating of Sonobuoy to match.

**Comment by xmudrii** [comment]
Pinging some folks involved with this PR, but also who are root-approvers...
@Riaankl @johnbelamaric @aojea @dims @liggitt Are we confident with this PR and do we want to remove the blocked-paths label? Same goes for #115510

**Review by xmudrii — APPROVED** [review]
For RelEng:
/lgtm
/approve

**Pull request merged**
